rotkehlchen/chain/ethereum/interfaces/ammswap/ammswap.py

```diff
@@ -0,0 +1,991 @@
+                        f'during {self.location} prices query from graph.'
+                    )
+                    log.error(msg)
+                    raise RemoteError(msg) from e
+                asset_price[token_address] = Price(FVal(tdd['priceUSD']))
+
+            # Check whether an extra request is needed
+            if len(result_data) < GRAPH_QUERY_LIMIT:
+                break
+
+            # Update pagination step
+            param_values = {
+                **param_values,
+                'offset': param_values['offset'] + GRAPH_QUERY_LIMIT,  # type: ignore
+            }
+
+        return asset_price
+
+    def get_events_history(
+        self,
+        addresses: List[ChecksumEthAddress],
+        reset_db_data: bool,
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+    ) -> AddressEventsBalances:
+        """Get the addresses' events history in the AMM"""
+        with self.trades_lock:
+            if reset_db_data is True:
+                self.delete_events_data()
+
+            address_events_balances = self._get_events_balances(
+                addresses=addresses,
+                from_timestamp=from_timestamp,
+                to_timestamp=to_timestamp,
+            )
+
+        return address_events_balances
+
+    def get_trades(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            only_cache: bool,
+    ) -> List[AMMTrade]:
+        if len(addresses) == 0:
+            return []
+
+        with self.trades_lock:
+            all_trades = []
+            trade_mapping = self._get_trades(
+                addresses=addresses,
+                from_timestamp=from_timestamp,
+                to_timestamp=to_timestamp,
+                only_cache=only_cache,
+            )
+            for _, trades in trade_mapping.items():
+                all_trades.extend(trades)
+
+            return all_trades
+
+    def _fetch_trades_from_db(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> AddressTrades:
+        """Fetch all DB AMM trades within the time range"""
+        db_address_trades: AddressTrades = {}
+        for address in addresses:
+            db_swaps = self.database.get_amm_swaps(
+                from_ts=from_timestamp,
+                to_ts=to_timestamp,
+                location=self.location,
+                address=address,
+            )
+            db_trades = self.swaps_to_trades(db_swaps)
+            if db_trades:
+                db_address_trades[address] = db_trades
+
+        return db_address_trades
+
+    def _get_balances_graph(
+            self,
+            addresses: List[ChecksumEthAddress],
+    ) -> ProtocolBalance:
+        """Get the addresses' pools data querying the Sushiswap subgraph
```

Comment:
Sushiswap should go from here.
```suggestion
        """Get the addresses' pools data querying the AMM's subgraph
```